evaluate/repository.go

```diff
@@ -80,3 +83,86 @@ func Repository(logger *log.Logger, resultPath string, model evalmodel.Model, la
 
 	return repositoryAssessment, problems, nil
 }
+
+// TemporaryRepository creates a temporary repository and initializes a git repo in it.
+func TemporaryRepository(logger *log.Logger, dataPath string) (temporaryRepositoryPath string, cleanup func(), err error) {
+	temporaryPath, err := os.MkdirTemp("", "eval-dev-quality")
+	if err != nil {
+		return "", cleanup, pkgerrors.WithStack(err)
+	}
+
+	cleanup = func() {
+		if e := os.RemoveAll(temporaryPath); e != nil {
+			if err != nil {
+				err = errors.Join(err, pkgerrors.WithStack(e))
+			} else {
+				err = pkgerrors.WithStack(e)
+			}
+		}
+	}
```

Comment:
Since we are returning the error above, does these `err` checks still apply? Or do the errors below play a role here?

Reply:
The `err` may be set inside the function where the cleanup function is getting deferred. So the `err` does not necessary reference the scope where the function is declared in.